controllers/operatorpolicy_status.go

```diff
@@ -552,6 +565,30 @@ func updatedCond(kind string) metav1.Condition {
 	}
 }
 
+type Schema string
+
+const (
+	Bundle  Schema = "Bundle"
+	Channel Schema = "Channel"
+	Package Schema = "Package"
+)
+
+func (s Schema) toString() string {
+	return string(s)
+}
+
+// deprecationCond returns a Compliant condition (without affecting the condition itself),
+// and a message in the format: 'The ____ (Package, Channel, Bundle) was deprecated.'
+// + additional details from packageManifest.
+func deprecationCond(depSchemaName string, depSchema Schema, message string) metav1.Condition {
```

Comment:
If we don't want this to affect the final compliance, we should ensure that this returns a `Status` that matches whatever is Compliant. Right now it doesn't.

To allow users to determine if they want the deprecation status to affect the overall compliance, we'll need to add something to the `ComplianceConfig`, and pass that into this function. For example, `buildDeploymentCond` uses that pattern.